src/integration/resources/inprocess/aggregator_test.go

```diff
@@ -0,0 +1,222 @@
+//go:build integration_v2
+// +build integration_v2
+
+// Copyright (c) 2021  Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package inprocess
+
+import (
+	"testing"
+
+	"github.com/m3db/m3/src/cluster/generated/proto/placementpb"
+	"github.com/m3db/m3/src/integration/resources"
+	"github.com/m3db/m3/src/msg/generated/proto/topicpb"
+	"github.com/m3db/m3/src/query/generated/proto/admin"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestNewAggregator(t *testing.T) {
```

Comment:
Can we add a test that restarts the agg in the same test? I have a similar update on one of my upcoming PRs, but it's a good way to ensure agg is always restartable in process.

Something like:
```go
func TestNewAggregator(t *testing.T) {
  // setup

  agg, err := NewAggregator(defaultAggregatorConfig, AggregatorOptions{})
  require.NoError(t, err)
  require.NoError(t, agg.Close())

  agg, err = NewAggregator(defaultAggregatorConfig, AggregatorOptions{})
  require.NoError(t, err)
  require.NoError(t, agg.Close())
}
```

Reply:
Done. But this test is gonna take long.

Reply:
Done. This test is taking long though.